[mix_model]: Fix issues and improve the lecture

bishmaybarik opened the pull request:
This PR fixes the following issues:

- Fixes #577

The issue was the following:

- [x] style guide review
- [x] fix full web links that are currently used across the lectures with a range of `link text`, `doc` links.

bishmaybarik commented:
hi @mmcky , I have made some changes and made sure I pushed everything to this branch. although I have checked it myself, it'll be nice to have it reviewed by you :-)

if there are further issues with the lecture, please let me know!

HumphreyYang commented on lectures/mix_model.md:
```diff
@@ -631,11 +628,11 @@ $$
 
-We'll assume that the person starts out with a prior probability $\pi_0(\alpha)$ on
+We'll assume that the agent starts out with a prior probability $\pi_0(\alpha)$ on
 $\alpha \in (0,1)$ where the prior has one of the forms that we deployed in {doc}`this quantecon lecture <bayes_nonconj>`.
```
```suggestion
$\alpha \in (0,1)$ where the prior has one of the forms that we deployed in {doc}`bayes_nonconj`.
```

HumphreyYang commented on lectures/mix_model.md:
```diff
@@ -33,32 +33,32 @@ A compound lottery can be said to create a _mixture distribution_.
 
 Our two ways of constructing a compound lottery will differ in their **timing**.
```
This might be out of the scope of this PR but we should only use bold for definitions

```suggestion
Our two ways of constructing a compound lottery will differ in their *timing*.
```

HumphreyYang commented on lectures/mix_model.md:
```diff
@@ -33,32 +33,32 @@ A compound lottery can be said to create a _mixture distribution_.
 
 $$
-\pi_t = E [ \textrm{nature chose distribution}  f | w^t] , \quad  t = 0, 1, 2, \ldots
+\pi_t = E [ \textrm{nature chose distribution} f | w^t] , \quad t = 0, 1, 2, \ldots
```
Adding a space between $f$ and the text
```suggestion
\pi_t = E [ \textrm{nature chose distribution } f | w^t] , \quad t = 0, 1, 2, \ldots
```

HumphreyYang commented on lectures/mix_model.md:
```diff
@@ -229,26 +229,24 @@ Here is pseudo code for a direct "method 1" for drawing from our compound lotter
 
-See <https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html> for the
-`searchsorted` function.
+See the [numpy.searchsorted documentation](https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html) for details on the `searchsorted` function.
```
```suggestion
See the [`numpy.searchsorted` documentation](https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html) for details on the `searchsorted` function.
```

HumphreyYang reviewed (commented):
Many thanks @bishmaybarik! These are great changes!

Noted that if we use 

```
{doc}`lecture_name`
```

It will give use a hyperlink with the lecture title. So we can reduce the usage of "this lecture" in the lectures.

There are also a few lines of code that's touching 80-character limit so please have a go at breaking them into two lines!

Please let me know what you think.

bishmaybarik committed:
[mix_model] Style-guide review: links, headings, prose

Rebased onto main. Fixes #577: converts bare/awkward links to MyST
{doc} and inline-link form, applies sentence-case section headings,
and cleans up prose/whitespace. Preserves the rng changes from #978.
